webapp/channels/src/actions/post_actions.ts

```diff
@@ -141,7 +147,28 @@ function storeCommentDraft(rootPostId: string, draft: null) {
 }
 
 export function addReaction(postId: string, emojiName: string) {
-    return (dispatch: DispatchFunc) => {
+    const getUniqueReactionsToPost = makeGetUniqueReactionsToPost();
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState() as GlobalState;
+        const config = getConfig(state);
+        const reactions = Object.values(getUniqueReactionsToPost(state, postId) ?? {});
+        let reactionCount = reactions.length;
+        if (reactions.some((reaction) => reaction.emoji_name === emojiName)) {
+            reactionCount--;
+        }
+
+        if (reactionCount >= Number(config.UniqueEmojiReactionLimitPerPost)) {
```

Comment:
Sorry I'm not sure what you're getting at here. Are we saying that we would assume that the limit on the post was already exceeded somehow? I think that would be an edge case and if the limit is exceeded then yes I think we should be preventing that because somewhere the data was corrupted.

Reply:
So imagine there's a post with 51 emojis, including `:wave:` -- I believe it's now impossible to add your own emoji to any of the existing emojis here, because even after we "subtract" `:wave:`, 50 >= 50, and we disallow (incorrectly, I believe).

Reply:
That would be correct actually, since 50 is the limit. 51 would be erroneous and bringing to 50 would "uncorrupt" the data. If we wanted to do another reaction, we would have to remove an existing one.

Reply:
I thought the limit was 50 /unique/ emojis, not 50 reactions. In this case, I'm just trying to add to the existing `:wave` check, but because this post happens to start over the limit, I'm blocked from doing so. I think that's wrong?

Reply:
I thought I tested it but I can have another look.

Reply:
Just ran a test with a post with 10 reactions on it (10 being the limit). Reduced the limit to 5, and I was required to remove 6 reactions (to get it down to 4) before it would let me add another one. So I think it's working fine, but I'll spin up a SpinWick here so you can test.

Reply:
Saw @harshilsharma63's comment in https://github.com/mattermost/mattermost/pull/25331#issuecomment-1822126765 and I think that's what I meant to describe -- adding to the existing reactions. Anyway, sounds like it's fixed now!

Reply:
Yeah, I thought that might be what you meant. Sometimes a video is all you need :rofl:
Anyways yes fixed, thanks for finding it!